Makefile

```diff
@@ -0,0 +1,27 @@
+.DEFAULT_GOAL := help
+
+# determines what "make help" will show
+define PRINT_HELP_PYSCRIPT
```

Comment:
never knew this could be done in Makefile!  I wouldn't  bother though and have a dedicated script under tools to simplify editing, avoiding double `$$` making troubleshooting more fun etc.

Reply:
This makefile is unlikely to survive till the end of this PR anyway.

For the ref, the idea comes from this python package cookiecutter 📦 🎂 ✂ 

https://github.com/audreyfeldroy/cookiecutter-pypackage/blob/master/%7B%7Bcookiecutter.project_slug%7D%7D/Makefile